parquet/src/main/java/org/apache/iceberg/parquet/ParquetSchemaUtil.java

```diff
@@ -164,4 +165,62 @@ public Boolean primitive(PrimitiveType primitive) {
     }
   }
 
+  public static Type determineListElementType(GroupType array) {
+    Type repeated = array.getFields().get(0);
+    boolean isOldListElementType = isOldListElementType(array);
+
+    Preconditions.checkArgument(isOldListElementType ||
+            repeated.asGroupType().getFieldCount() <= 1,
+        "Invalid list: repeated group is not a single field: %s", array);
```

Comment:
This will never happen because if the field count is > 1, then `isOldListElementType` will be true.

Reply:
Good point, I probably got this while refactoring `ParquetTypeVisitor`. Dropping, thanks.